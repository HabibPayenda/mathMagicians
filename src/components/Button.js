/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import './styles.css';
import React from 'react';

class Button extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    const {
      large, color, text, changeCalculation,
    } = this.props;
    return (
      <div data-testid="button" className={`button ${color} ${large}`} onClick={() => changeCalculation(text)}>
        {text}
      </div>
    );
  }
}

export default Button;
