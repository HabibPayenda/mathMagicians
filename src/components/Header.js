import './styles.css';
import React from 'react';

class Header extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    const { total, next, operation } = this.props;
    return (
      <div data-testid="header" className="header">
        {total}
        {operation}
        { next }
      </div>
    );
  }
}

export default Header;
