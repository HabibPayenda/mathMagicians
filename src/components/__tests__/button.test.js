import { screen, render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import reactTestRenderer from 'react-test-renderer';

import Button from '../Button';
import Header from '../Header';

describe('Button tests', () => {
  it('should see the element in the dom tree', () => {
    const buttonName = '-';

    render(<Button text={buttonName} />);

    const buttonElement = screen.getByTestId('button');
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveTextContent('-');
  });

  it('should match the snapshot', () => {
    const buttonName = '-';
    const tree = reactTestRenderer.create(<Button text={buttonName} />);
    expect(tree).toMatchSnapshot();
  });

  it('should check if the click event is working properly', () => {
    const buttonName = '1';

    const changeCalculation = jest.fn();

    render(<Button text={buttonName} changeCalculation={changeCalculation} />);
    render(<Header total={buttonName} />);

    const buttonElement = screen.getByTestId('button');
    const headerElement = screen.getByTestId('header');
    fireEvent.click(buttonElement);
    expect(headerElement.textContent).toBe('1');
  });
});
