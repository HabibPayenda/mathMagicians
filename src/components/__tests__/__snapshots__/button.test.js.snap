// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests should match the snapshot 1`] = `
<div
  className="button undefined undefined"
  data-testid="button"
  onClick={[Function]}
>
  -
</div>
`;

exports[`should match the snapshot 1`] = `
<div
  className="button undefined undefined"
  data-testid="button"
  onClick={[Function]}
>
  -
</div>
`;
